### PR TITLE
Add new custom services

### PR DESCRIPTION
**Pull request by ahmam**
add tne custom service when the package installed from custom Debian repository.

### CODE REVIEW

File: tasks/slurmctld.yml [hunk 1]
```diff
@@ -25,6 +25,17 @@
     state: directory
   when: slurm_create_dirs and __slurm_config_merged.SlurmctldLogFile != omit
 
+- name: add slurmctld service
```
Comment (btravouillon) [inline tasks/slurmctld.yml]:
```suggestion
- name: Add slurmctld service
```

File: tasks/slurmd.yml [hunk 2]
```diff
@@ -24,6 +24,17 @@
     mode: 0755
     state: directory
   when: slurm_create_dirs and __slurm_config_merged.SlurmdLogFile != omit
+ 
+- name: add slurmd service
```
Comment (btravouillon) [inline tasks/slurmd.yml]:
```suggestion
- name: Add slurmd service
```

**Review by btravouillon — COMMENTED** [review]
The three j2 files do not look like Jinja2 template. These are plain files with no Ansible variable, so you should use the module `copy` instead of `template`.

https://www.redhat.com/sysadmin/ansibles-copy-template-modules

**Comment by ahmam** [comment]
I started with the idea of adding a binary directory variable in order to leave the choice of installation ( `{{ slurm_path }}` et `{{ slurmd_sysconfig }}` ) anyway i modify with copy module

**Comment by ahmam** [comment]
fixed

File: tasks/slurmctld.yml [hunk 3]
```diff
@@ -25,6 +25,17 @@
...
   when: slurm_create_dirs and __slurm_config_merged.SlurmctldLogFile != omit
 
+- name: Add slurmctld service
+  ansible.builtin.copy:
+    src: slurmctld.service
+    dest: /etc/systemd/system/slurmctld.service
+    owner: root
+    group: root
+    mode: 0644
+  when: slurm_configure_repos
```
Comment (btravouillon) [inline tasks/slurmctld.yml]: I wonder if we should add a new variable `slurm_configure_systemd` (default `false`) to setup systemd unit files.

It is legit to use a custom repo with packages that include the systemd unit files. Thus, there should be no correlation between using a custom repo and having the systemd unit files installed by the role.

Comment (ahmam) [reply tasks/slurmctld.yml]: i added this new variable ( slurm_configure_systemd ) to connection systemd unit files and this condition

File: tasks/slurmctld.yml [hunk 4]
```diff
@@ -25,6 +25,17 @@
...
+- name: Add slurmctld service
+  ansible.builtin.copy:
+    src: slurmctld.service
+    dest: /etc/systemd/system/slurmctld.service
+    owner: root
+    group: root
+    mode: 0644
+  when: slurm_configure_systemd
+  notify:
+    - restart slurmctld
```
Comment (btravouillon) [inline tasks/slurmctld.yml]: Thoe notify tasks start witrh a capital `R` in https://github.com/mila-iqia/ansible-slurm/blob/mila/handlers/main.yml

Comment (ahmam) [reply tasks/slurmctld.yml]: fixed

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

File: tasks/slurmctld.yml (hunk 4 above)
Comment (ahmam) [reply tasks/slurmctld.yml]: I just modified the service module by systemd for the restart of service at the end of reloading the systemd deamon

File: handlers/main.yml [hunk 5]
```diff
@@ -11,9 +11,12 @@
   when: "slurm_start_services and ('slurmexechosts' in group_names or 'exec' in slurm_roles)"
 
 - name: Restart slurmd
-  ansible.builtin.service:
+  ansible.builtin.systemd:
     name: "{{ slurmd_service_name }}"
     state: restarted
+    masked: no
+    enabled: yes
+    daemon_reloas: yes
```
Comment (btravouillon) [inline handlers/main.yml]:
```suggestion
    daemon_reload: yes    
```

Comment (btravouillon) [inline handlers/main.yml]:
```suggestion
    daemon_reloas: yes
```

File: tasks/slurmd.yml [hunk 6]
```diff
@@ -24,6 +24,17 @@
     mode: 0755
     state: directory
   when: slurm_create_dirs and __slurm_config_merged.SlurmdLogFile != omit
+
```
Comment (btravouillon) [inline tasks/slurmd.yml]:
```suggestion

```

File: handlers/main.yml [hunk 7]
```diff
@@ -23,11 +26,23 @@
...
 
+- name: restart slurmdbd
+  ansible.builtin.systemd:
+    name: "{{ slurmdbd_service_name }}"
+    state: restarted
+    masked: no
+    enabled: yes
+    daemon_reload: yes
+  when: "slurm_start_services and ('slurmservers' in group_names or 'controller' in slurm_roles)" 
+
```
Comment (btravouillon) [inline handlers/main.yml]:
```suggestion
- name: Restart slurmdbd
  ansible.builtin.systemd:
    name: "{{ slurmdbd_service_name }}"
    state: restarted
    masked: no
    enabled: yes
    daemon_reload: yes
  when: "slurm_start_services and ('slurmservers' in group_names or 'controller' in slurm_roles)"

```

**Review by btravouillon — CHANGES_REQUESTED** [review]

File: handlers/main.yml (hunk 5 above)
Comment (ahmam) [reply handlers/main.yml]: fixed

File: tasks/slurmd.yml [hunk 8]
```diff
@@ -24,6 +24,16 @@
     mode: 0755
     state: directory
   when: slurm_create_dirs and __slurm_config_merged.SlurmdLogFile != omit
+- name: Add slurmd service
```
Comment (btravouillon) [inline tasks/slurmd.yml:28]:
```suggestion

- name: Add slurmd service
```

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Review by pbouchez — APPROVED** [review]

**Pull request merged**